abi.go

```diff
@@ -0,0 +1,170 @@
+package ebpf
+
+import (
+	"github.com/pkg/errors"
+)
+
+// CollectionABI describes the interface of an eBPF collection.
+type CollectionABI struct {
+	Maps     map[string]*MapABI
+	Programs map[string]*ProgramABI
```

Comment:
Could we somehow add an ABI for the load rewriting too? Would be cool to be able to declare upfront a full `CollectionABI` with everything.

Reply:
Not sure what you mean.

Reply:
If the ABI stuff is only for loading pinned programs this doesn't make much sense.